partner_other_names/models/res_partner.py

```diff
@@ -0,0 +1,128 @@
+                names.append(lastname2)
+            if names and (firstname or othernames) and order == 'last_first_comma':
+                names[-1] = names[-1] + ","
+            if firstname:
+                names.append(firstname)
+            if othernames:
+                names.append(othernames)
+        return u" ".join(names)
+
+    @api.depends("firstname", "othernames", "lastname", "lastname2")
+    def _compute_name(self):
+        """Write the 'name' according to splitted data."""
+        for partner in self:
+            partner.name = self._get_computed_name(
+                partner.firstname,
+                partner.othernames,
+                partner.lastname,
+                partner.lastname2)
+
+    @api.model
+    def _names_order_default(self):
+        return 'first_last'
+
+    @api.multi
+    def _inverse_name(self):
+        """Try to revert the effect of '_compute_name'."""
+        for record in self:
+            parts = record._get_inverse_name(record.name, record.is_company)
+            record.lastname = parts['lastname']
+            record.lastname2 = parts['lastname2']
+            record.firstname = parts['firstname']
+            record.othernames = parts['othernames']
+
+    @api.model
+    def _get_inverse_name(self, name, is_company=False):
+        """
+        Compute the inverted name.
+        - If the partner is a company, save it in the lastname.
+        - Otherwise, make a guess.
+        """
+        # Company name goes to the lastname
+        result = {
+            'firstname': False,
+            'othernames': False,
+            'lastname': name or False,
+            'lastname2': False,
+        }
+
+        if not is_company and name:
+            order = self._get_names_order()
+            result = super(ResPartner, self)._get_inverse_name(name, is_company)
+            parts = []
+
+            if order == 'first_last':
+                if result['lastname2']:
+                    parts = result['lastname2'].split(" ", 1)
+                while len(parts) < 2:
+                    result['othernames'] = False
+                    return result
+                result['othernames'] = result['lastname']
+                result['lastname'] = parts[0]
+                result['lastname2'] = parts[1]
+            else:
+                if result['firstname']:
+                    parts = result['firstname'].split(" ", 1)
+                while len(parts) < 2:
+                    parts.append(False)
+                result['firstname'] = parts[0]
+                result['othernames'] = parts[1]
+
+        return result
+
+    @api.constrains("firstname", "othernames", "lastname", "lastname2")
+    def _check_name(self):
+        """Ensure at least one name is set."""
+        try:
+            super(ResPartner, self)._check_name()
+        except exceptions.EmptyNamesError:
+            for partner in self:
+                if not partner.othernames:
+                    raise
```

Comment:
This can do the wrong thing if there are multiple partners.
If we run it on partner A and B, where A only has a `firstname` and B only has an `othernames`, then it raises because `A` doesn't have an `othernames` even though that's fine.
I think this would work better:
```suggestion
        for partner in self:
            try:
                super(ResPartner, partner)._check_name()
            except exceptions.EmptyNamesError:
                if not partner.othernames:
                    raise
```

Reply:
TY for your revision: you're right, it was fixed.